packages/node/src/methods/app-instance-operations.ts

```diff
@@ -1,25 +1,72 @@
-import { Node } from "@counterfactual/types";
+import { AppInstance, StateChannel } from "@counterfactual/machine";
+import { AppInstanceInfo, Node } from "@counterfactual/types";
 
-import { APP_INSTANCE_STATUS } from "../db-schema";
+import { Store } from "../store";
 
 import { RequestHandler } from "./request-handler";
 
-export async function getInstalledAppInstances(
+export async function getProposedAppInstances(
   this: RequestHandler
 ): Promise<Node.GetAppInstancesResult> {
   return {
-    appInstances: await this.channels.getAppInstances(
-      APP_INSTANCE_STATUS.INSTALLED
-    )
+    appInstances: await this.store.getProposedAppInstances()
   };
 }
 
-export async function getProposedAppInstances(
+/**
+ * Gets all installed appInstances across all of the channels open on
+ * this Node.
+ */
+export async function getInstalledAppInstances(
   this: RequestHandler
 ): Promise<Node.GetAppInstancesResult> {
+  const appInstances: AppInstanceInfo[] = [];
+  const channels = await this.store.getAllChannels();
+  for (const channel of Object.values(channels)) {
+    if (channel.appInstances) {
+      const nonFreeBalanceAppInstances = getNonFreeBalanceAppInstancesJSON(
+        channel
+      );
+      const appInstanceInfos = await getAppInstanceInfoFromAppInstance(
+        this.store,
+        nonFreeBalanceAppInstances
+      );
+      appInstances.push(...Object.values(appInstanceInfos));
+    } else {
+      console.log(
+        `No app instances exist for channel with multisig address: ${
+          channel.multisigAddress
+        }`
+      );
+    }
+  }
   return {
-    appInstances: await this.channels.getAppInstances(
-      APP_INSTANCE_STATUS.PROPOSED
-    )
+    appInstances
   };
 }
+
+async function getAppInstanceInfoFromAppInstance(
+  store: Store,
+  appInstances: AppInstance[]
+): Promise<AppInstanceInfo[]> {
+  return await Promise.all(
+    appInstances.map<Promise<AppInstanceInfo>>(
+      async (appInstance: AppInstance): Promise<AppInstanceInfo> => {
+        const appInstanceId = await store.getAppInstanceIDFromAppInstanceIdentityHash(
+          appInstance.identityHash
+        );
+        return await store.getAppInstanceInfo(appInstanceId);
+      }
+    )
+  );
+}
+
+function getNonFreeBalanceAppInstancesJSON(
+  stateChannel: StateChannel
+): AppInstance[] {
+  return [...stateChannel.appInstances.values()].filter(
```

Comment:
```suggestion
  return stateChannel.appInstances.values().filter(
```

Reply:
`.values()` returns an `Iterator`, which can't be `filter`ed.

Reply:
Oh wow, weird. I would have thought `Map` would have a `filter` method.